Dronuts/client_src/src/components/customer/Menu.js

```diff
@@ -1,112 +1,141 @@
 import React, {Component} from 'react';
 import ReactDOM from 'react-dom';
+import axios from 'axios';
 
-var cart = []
+class Menu extends Component {
+  constructor(){
+    super();
+    this.state = {
+      menu: [],
+      cart: []
+    };
+  }
 
-function addToCart(item){
-  var itemInCart = cart.find((cartItem) => (cartItem.id === item.id));
-  if (itemInCart == null) {
-    cart.push({
-      "id" : item.id,
-      "name" : item.name,
-      "price" : item.price,
-      "quantity" : 1
-    });
-  } else {
-    itemInCart.quantity++;
+  componentDidMount(){
+    axios.get('/api/Items')
+      .then(response => {
+        this.setState({menu: response.data});
+      })
+
+    if(sessionStorage.getItem('cart')){
+      this.setState({cart: JSON.parse(sessionStorage.getItem('cart'))});
+    }
   }
 
-  const cartRows = cart.map((cartItem) => (
-    <div className="row">
-      <div className="col-5 border">
-        {cartItem.name}
-      </div>
-      <div className="col-4 border text-right">
-        ${cartItem.price}
-      </div>
-      <div className="col-3 border text-right">
-        {cartItem.quantity}
-      </div>
-    </div>
-  ));
+  addToCart(item){
+    var cart = this.state.cart;
+    var itemInCart = cart.find((cartItem) => (cartItem.id === item.id));
+    if (itemInCart == null) {
+      cart.push({
+        "id" : item.id,
+        "name" : item.name,
+        "price" : item.price,
+        "quantity" : 1
+      });
+    } else {
+      itemInCart.quantity++;
+    }
+    this.setState({cart: cart});
+    sessionStorage.setItem('cart', JSON.stringify(this.state.cart));
+  }
 
-  var total = 0;
-  cart.map((cartItem) => (
-    total += cartItem.price * cartItem.quantity
-  ));
-  total = Math.round(total * 100) / 100
+  removeFromCart(item){
+    var cart = this.state.cart;
+    var itemInCart = cart.find((cartItem) => (cartItem.id === item.id));
+    if (itemInCart == null) return;
+    if (itemInCart.quantity <= 1) {
+      cart.splice(cart.indexOf(itemInCart), 1);
+    } else {
+      itemInCart.quantity--;
+    }
+    this.setState({cart: cart});
```

Comment:
This will become an issue when I PR cause I preserved only the order $ total, not the whole cart in session storage. I agree though that saving the whole cart makes more sense so we can add the items to the orders data model later on.

Reply:
We can deal with that as we try to merge your PR.